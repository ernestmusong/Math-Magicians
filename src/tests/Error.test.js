import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import Error from '../components/Error';

describe('Error', () => {
  test('should render the Error component', () => {
    const tree = renderer.create(<Error />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should have a text of Ooops!', () => {
    render(<Error />);
    const errorElement = screen.getByText(/Oops!/i);
    expect(errorElement).toBeInTheDocument();
  });
});
