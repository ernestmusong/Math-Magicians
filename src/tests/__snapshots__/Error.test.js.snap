// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error should render the Error component 1`] = `
<div
  id="error-page"
>
  <h1>
    Oops!
  </h1>
  <p>
    Sorry, an unexpected error has occurred.
  </p>
  <p>
    <i>
      Page not found
    </i>
  </p>
</div>
`;
